homeassistant/components/guardian/__init__.py

```diff
@@ -142,50 +257,131 @@ def icon(self) -> str:
         """Return the icon."""
         return self._icon
 
+    @callback
+    def _async_update_from_latest_data(self):
+        """Update the entity.
+
+        This should be extended by Guardian platforms.
+        """
+        raise NotImplementedError
+
+    @callback
+    def _async_update_state_callback(self):
+        """Update the entity's state."""
+        self._async_update_from_latest_data()
+        self.async_write_ha_state()
+
+
+class PairedSensorEntity(GuardianEntity):
```

Comment:
Why not inherit from `CoordinatorEntity`?

Reply:
`GuardianEntity` inherits from `CoordinatorEntity`.